### PR TITLE
Add support for classes nested in objects

### PR DESCRIPTION
**Pull request by mrobakowski**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,13 +41,13 @@ The plugin is currently published for the following Scala versions:
 
 1. Only case classes located directly in `package`s or `object`s are changed. Nested classes and classes local to functions are currently ignored.
 2. Only the fields in the first parameter list are shown.
-3. If the class already overrides `toString` *directly*, it's not replaced.
+3. If the class is nested in an object (but not a package object), its name and a dot are prepended.
```
Comment (mrobakowski) [inline README.md:44]:
```suggestion
3. If the class is nested in an object (but not a package object), the object's name and a dot are prepended.
```

**Comment by kubukoz** [comment]
I'll probably look at it after the weekend :D nice

**Comment by kubukoz** [comment]
Note to self: readme / implementation of the logic seems good, need to look closer at compilerapi implementations

**Comment by kubukoz** [comment]
also: #29 goes first so we can release the next patch version, this one will be a minor bump so should be merged later ;)

**Comment by mrobakowski** [comment]
> Note to self: readme / implementation of the logic seems good, need to look closer at compilerapi implementations

Btw, unfortunately, the code responsible for finding the enclosing object isn't in compilerapi, but in the respective plugins instead. That's because of the Scala 2 limitation that you found earlier (the finding parent / traversal thingie).

**Comment by mrobakowski** [comment]
Also, I think I'll rename `ClazzParent` to `EnclosingObject` as that is a bit closer to the truth.

**Comment by kubukoz** [comment]
riiiiiight. I'll look into that anyway, maybe it can be made cleaner

**Comment by kubukoz** [comment]
The conflict is just formatting, you can rebase with `git checkout --theirs` and reformat again :)

File: tests/src/test/scala/Demo.scala [hunk 2]
```diff
@@ -63,6 +75,10 @@ final class NestedParent() {
   case class NestedChild(name: String)
 }
 
+object ObjectNestedParent {
```
Comment (kubukoz) [inline tests/src/test/scala/Demo.scala:78]: Gonna need a test case where this object is in a class - in that case we shouldn't be changing the `toString` at all (I want to keep consistent behavior for things nested in classes for now)

Comment (mrobakowski) [reply tests/src/test/scala/Demo.scala:84]: Just added that in https://github.com/kubukoz/better-tostring/pull/27/commits/7ac685015a1d49bc5a9acbbe773d0eeadd42f7e6

**Comment by kubukoz** [comment]
Made some simplifications where I could find some, tests still pass so I think it's going to be fine :D 

If you think this is fine @mrobakowski (and CI passes) I'm good to merge

**Comment by kubukoz** [comment]
damn, our CI is just too fast ngl

**Review by kubukoz — APPROVED** [review]

**Comment by mrobakowski** [comment]
Nice! I'm gonna merge this then.

**Pull request merged**